src/parser.rs

```diff
@@ -145,6 +145,29 @@ osd-ldiskfs.fs2-MDT0000.kbytestotal=1819968
     fn test_node_output() {
         let x = include_str!("./fixtures/valid/valid.txt");
 
+        let result = parse()
+            .easy_parse(x)
+            .map_err(|err| err.map_position(|p| p.translate_position(x)))
+            .unwrap();
+
+        assert_debug_snapshot!(result);
+    }
+    #[test]
+    fn test_node_output_single_valid() {
+        let x = include_str!("./fixtures/valid/valid-single.txt");
+
+        let result = parse()
+            .easy_parse(x)
+            .map_err(|err| err.map_position(|p| p.translate_position(x)))
+            .unwrap();
+
+        assert_debug_snapshot!(result);
+    }
+
+    #[test]
+    fn test_node_output_should_be_valid() {
```

Comment:
This throws:
```
thread 'parser::tests::test_node_output_should_be_valid' panicked at 'called `Result::unwrap()` on an `Err` value: Errors { position: 321, errors: [Unexpected(Token('a')), Expected(Static("stats")), Message(Static("while parsing export_stats")), Message(Static("while parsing obdfilter"))] }', src/parser.rs:174:14
```

Reply:
I was able to solve it by wrapping the parser around `attempt`.  see https://github.com/whamcloud/lustre-collector/pull/57/files#diff-e53daf17f57ad4f274ee3f785a28bd0e488687144836c67bff86e839c67489b7R78-R90
It still unclear to me why this solve the issue though.